/**
 * FILE: Core.proto
 * BRIEF: Main protocol file for the SOAR Avionics System, any shared enums or messages should be defined here.
 * AUTHOR: Christopher Chan (cjchanx)
 */
syntax = "proto3";
package Proto;

/* Universal Enums ------------------------------------------------------------------*/

/* These are the nodes in the SOAR Avionics System.
 * The values are used to identify the source or destination of a message.
*/
enum Node {
    NODE_INVALID = 0;

    NODE_ANY = 1;
    NODE_RCU = 2;
    NODE_DMB = 3;
    NODE_PBB = 4;
}

enum MessageID {
    MSG_INVALID = 0;

    MSG_CONTROL = 1;
    MSG_COMMAND = 2;
    MSG_TELEMETRY = 3;
}

/* Board Specific Enums ------------------------------------------------------------------*/

enum RocketState {
    DMB_INVALID = 0;

    // Current DMB RocketState (these should be kept close to the actual RocketState enum in RocketSM.hpp)
    RS_PRELAUNCH = 1;
    RS_FILL = 2;
    RS_ARM = 3;
    RS_IGNITION = 4;
    RS_LAUNCH = 5;
    RS_BURN = 6;
    RS_COAST = 7;
    RS_DESCENT = 8;
    RS_RECOVERY = 9;
    RS_ABORT = 10;
    RS_NONE = 11;
}